ui/src/FindBar.tsx

```diff
@@ -1,23 +1,53 @@
 import React from "react";
+import { convertToAnnotationId } from './selectors/annotation'
 import { EventBus } from "./types/pdfjs-viewer";
+import { ScholarReaderContext } from "./state";
+import { BoundingBox } from "./types/api";
+
+interface FindBarProps {
+  mappingToBounds: Map<String, BoundingBox>;
+  matches: string[];
+  selectedSymbol: string;
+}
 
 interface FindBarState {
   currentMatch: number | null;
   matchCount: number | null;
+  mode: string,
   /**
    * Event bus for the pdf.js application. The logic for performing text search within the PDF is
    * provided by pdf.js. This event bus is needed for communicating with the pdf.js code.
    */
   pdfJsEventBus?: EventBus;
 }
 
-export class FindBar extends React.PureComponent<{}, FindBarState> {
-  constructor(props: {}) {
-    super(props);
-    this.state = {
-      currentMatch: null,
-      matchCount: null,
-    };
+const initialState = {
+  currentMatch: null,
+  matchCount: null,
+  mode: 'hidden',
+}
+
+export class FindBar extends React.PureComponent<FindBarProps, FindBarState> {
+  state = initialState;
```

Comment:
Is it a better idiom in React to set state in this way vs. in a constructor?

Reply:
Also, can we be sure that the `initalState` object isn't changed by later calls to `setState` within the component?

Reply:
Yes, if the only thing in your constructor is state then it's best to not have the constructor (here is a [blog post](https://overreacted.io/why-do-we-write-super-props/) about it from one of the core maintainers of react if you are curious!).

Reply:
Since state is immutable I believe that this should be safe